azure-pipelines.yml

```diff
@@ -216,8 +216,23 @@ stages:
         python3 packaging/change_name.py pyproject.toml "NMODL${TAG}"
         export SETUPTOOLS_SCM_PRETEND_VERSION="$(git describe --tags | cut -d '-' -f 1,2 | tr - .)"
         CIBW_ENVIRONMENT_PASS_LINUX='SETUPTOOLS_SCM_PRETEND_VERSION' python3 -m cibuildwheel --output-dir wheelhouse
-      condition: succeeded()
+      condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
       displayName: 'Building ManyLinux Wheels'
+    - script: |
+        if [[ "$(RELEASEWHEELBUILD)" != "True" ]]; then
+          export TAG="-nightly"
+        else
+          export TAG=""
+        fi
+        python3 -m pip install --upgrade pip
+        python3 -m pip install cibuildwheel==2.16.5 tomli tomli-w
+        # change the name accordingly
+        python3 packaging/change_name.py pyproject.toml "NMODL${TAG}"
+        export SETUPTOOLS_SCM_PRETEND_VERSION="$(git describe --tags | cut -d '-' -f 1,2 | tr - .)"
+        CIBW_BUILD="cp38* cp312*" CIBW_ENVIRONMENT_PASS_LINUX='SETUPTOOLS_SCM_PRETEND_VERSION' python3 -m cibuildwheel --output-dir wheelhouse
```

Comment:
Can one reduce the duplication by setting `CIBW_BUILD` to one value for nightly and another value for regular PR CI runs?

Reply:
I think I removed most of the duplication now, with two things to keep in mind (i.e. an explanation why it works):
- any azure variable not set, but used later (via `$(VARIABLE)`) will expand to an empty string
- if `CIBW_BUILD` is set to the empty string, `cibuildwheel` will ignore it and build all configurations (which is what we want for non-PRs)